File: lab_python_oop/circle.py
from lab_python_oop.shape import GeometricShape
from lab_python_oop.colour import Colour
from math import pi

class Circle(GeometricShape):

    FIGURE_TYPE = "Круг"

    @classmethod
    def getFigureType(class_v):
        return class_v.FIGURE_TYPE

    def __init__(self, r, col):
        self._radius = r
        self._colour = Colour()
        self._colour.shapeColour = col

    @property
    def Radius(self):
        return self._radius

    @Radius.setter
    def Radius(self, r):
        self._radius = r

    @Radius.deleter
    def Radius(self, r):
        del self._radius

    def Area(self):
        return pi * (self.Radius * self.Radius)

    def __repr__(self):
        return '{} c радиусом {} и площадью {}'.format(
            Circle.getFigureType(),
            self.Radius,
            self.Area()
        )

File: figures/circle.cs
namespace class_lab3
{
    /// <summary>
    /// Круг 
    /// </summary>
    class Circle : GeometricShape
    {
        /// <summary>
        /// Конструктор по радиусу
        /// </summary>
        /// <param name="r">Радиус круга</param>
        public Circle(double r)
        {
            Radius = r;
        }
        /// <summary>
        /// Радиус круга
        /// </summary>
        public double Radius { get; set; }

        public override double Area()
        {
            return Math.PI * Radius * Radius;
        }
        public override string ToString()
        {
            return ("Круг с радиусом " + Radius.ToString() + "; площадь: " + this.Area());
        }
    }
}